pkg/gardenlet/controller/managedseed/managedseed_actuator.go

```diff
@@ -151,7 +147,18 @@ func (a *actuator) Reconcile(ctx context.Context, ms *seedmanagementv1alpha1.Man
 		}
 	}
 
-	updateCondition(status, seedmanagementv1alpha1.ManagedSeedSeedRegistered, gardencorev1beta1.ConditionTrue, gardencorev1beta1.EventReconciled, "")
+	seed, err := a.getSeed(ctx, ms)
```

Comment:
Hmm, currently it is not checked if the Seed object exists for determining the ManagedSeed's `SeedRegistered` condition, correct?
So if I see it correctly, this PR/commit is mixing a bugfix and a change in the reconciler/condition logic.
If so, I would propose to split things up into multiple PRs, because I would `/lgtm` the bugfix but would like to discuss the logic change.

Reply:
I agree with @timebertt . The current behavior of the `SeedRegistered` condition is used by the `ManagedSeedSet` actuator, so it's rather important that he logic itself is not changed, both during reconcile and delete. Instead, I would try to limit the change to `updateCondition` and the subsequent status update (`updateStatus` in the reconciler) to ensure that the timestamps in the status are updated only if there is an actual change to the status.

Reply:
One way to achieve the above would be to change the `defer func` here: https://github.com/gardener/gardener/blob/b12f41152e867e6c0678311c9a82400ea7f0e5e6/pkg/gardenlet/controller/managedseed/managedseed_actuator.go#L87-L92 into something like:

```go
	defer func() {
		if err != nil {
			seedRegisteredStatus, seedRegisteredReason, seedRegisteredMessage = gardencorev1beta1.ConditionFalse, gardencorev1beta1.EventReconcileError, err.Error()
			a.reconcileErrorEventf(ms, err.Error())
		}
		updateCondition(status, seedmanagementv1alpha1.ManagedSeedSeedRegistered, seedRegisteredStatus, seedRegisteredReason, seedRegisteredMessage)
	}()
```

and then replace the other `updateCondition` calls with similar `seedRegisteredStatus, seedRegisteredReason, seedRegisteredMessage = ...`

In this way, there will be only one `updateCondition` call just before exiting the method. Something similar should work for `Delete` as well.